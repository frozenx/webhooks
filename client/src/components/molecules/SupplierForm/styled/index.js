import styled from 'styled-components'
import { media } from '@beans/foundation';
import { IconButtonLinkWithText } from '@beans/link';



export const SupplierStepThree = styled.div``;

export const BackWrapper = styled.div`
    display: flex;
    align-items:center;
    border-top: 1px solid #ccc;
    border-bottom: 1px solid #ccc;
    height: 56px;
    width: 1236px;
    @media (min-width: 1260px) {
        width: 1416px;
    }
    margin: 24px auto 10px auto;
    
`
export const StyledFormContainer = styled.div`
    margin-bottom : 24px;
`