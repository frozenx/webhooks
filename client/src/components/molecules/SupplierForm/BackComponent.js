import React from "react";
import { IconButtonLinkWithText } from '@beans/link';
import { BackWrapper } from "./styled";
import { collectionKeys } from '../../../config';

const BackComponent = ({ handleBackComponent, selectedAttr }) => {
    return (
        <BackWrapper>
            <IconButtonLinkWithText
                icon={{ graphic: 'backwardLink', size: 'sm', position: { global: 'left' } }}
                buttonVariant="secondary"
                emphasized={true}
                onClick={() => handleBackComponent(selectedAttr)}
            >
                {
                    selectedAttr && selectedAttr.collectionType ?
                        <span>{collectionKeys[selectedAttr.collectionType]}</span>
                        :
                        <span>{collectionKeys.default}</span>
                }
            </IconButtonLinkWithText>
        </BackWrapper>
    );
}

export default BackComponent;