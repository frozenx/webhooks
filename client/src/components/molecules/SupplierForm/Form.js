import React from "react";
import GenericFormContainer from "../../molecules/GenericForm/container";
import {
    BtnContainer,
} from '../../../common/styles'
import { StyledFormContainer } from "./styled";

import Button from '@beans/button';
import i18data from '../../../languagepack';

const SupplierDetailForm = props => {
    const {
        attributeToRulesMapping,
        genericFormData,
        id,
        onSave,
        values,
        errors,
        isFormLocked,
        cancel,
        isCancelModelOpen,
        isOnEditMode,
    } = props;
    return (
        <StyledFormContainer>
            <GenericFormContainer
                attributeToRulesMapping={attributeToRulesMapping}
                genericFormData={genericFormData}
                id={id}
                values={values}
                errors={errors}
                isFormLocked={isFormLocked}
                isOnEditMode={isOnEditMode}
            />
            <BtnContainer maxWidth={Math.max(i18data.save.length, i18data.cancel.length)}>
                < Button
                    variant="secondary"
                    className="submit-btn"
                    id="save-as-draft"
                    type="submit"
                    disabled={isFormLocked}
                    onClick={(e) => { e.stopPropagation(); e.preventDefault(); cancel(e) }}
                > <span>{i18data.cancel}</span>
                </Button>
                <Button
                    variant="primary"
                    className={`submit - btn enabled`}
                    type="submit"
                    id='save'
                    disabled={isFormLocked}
                    onClick={(e) => { e.stopPropagation(); e.preventDefault(); onSave(id); }}
                > <span>{i18data.save}</span>
                </Button>
                {/* <CustomModal
                    stayText={i18data.no}
                    cancelText={i18data.yesCancel}
                    text={i18data.cancelFormText}
                    isOpen={isCancelModelOpen}
                    stayForm={() => cancel('stay')} 
                    cancelModel={() => cancel('cancel', id)}
                />                 */}
            </BtnContainer >
        </StyledFormContainer >
    );
}

export default SupplierDetailForm;
