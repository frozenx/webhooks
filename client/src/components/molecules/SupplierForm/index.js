import React, { useState } from "react";
import Button from '@beans/button';
import Notification from '@beans/notification';
import { FormHeading, PrimaryWrapper, SecondaryWrapper, FormHeadingWrapper, NotificationStyle } from "../../../common/styles";
import SupplierDetailForm from './Form'
import { SupplierStepThree } from "./styled";
import BackComponent from './BackComponent'

const SupplierForm = ({
  attributeToRulesMapping,
  selectedAttr,
  partnerName,
  handleBackComponent,
  isFormValidationError,
  isFormSubmissionError,
  onSave,
  values,
  errors,
  node,
  cancel,
  isCancelModelOpen,
}) => {
  let currentHeaderName = selectedAttr.name;
  const currentFieldAction = node.fields.find(field => field.key === selectedAttr.key).action;
  let isEditMode = currentFieldAction === 'enabled' && Object.keys(values).length > 0;
  const [isFormLocked, setFormLock] = useState(isEditMode);

  return (
    <SupplierStepThree>
      <BackComponent handleBackComponent={handleBackComponent} selectedAttr={selectedAttr} />
      <PrimaryWrapper>

        <SecondaryWrapper>
          {(isFormSubmissionError || isFormValidationError) &&
            <NotificationStyle>
              <Notification title="We were unable to submit the form" variant="error">
                {isFormSubmissionError && 'Looks like something went wrong. Please try submitting again'}
                {isFormValidationError && 'Please complete the highlighted fields to continue'}
              </Notification>
            </NotificationStyle>
          }
          <FormHeadingWrapper>
            <FormHeading><h1>{`${partnerName} ${currentHeaderName}`}</h1></FormHeading>
            {isFormLocked &&
              <Button variant='secondary'
                onClick={() => setFormLock(false)}
                disabled={isEditMode && !node.attributeRules[node.type][selectedAttr.key].update}
              >
                Edit details
                </Button>
            }
          </FormHeadingWrapper>
          <SupplierDetailForm
            attributeToRulesMapping={attributeToRulesMapping}
            id={selectedAttr.key}
            genericFormData={selectedAttr.attributes}
            key={`selectedAttr-${selectedAttr.id}`}
            onSave={() => { onSave(selectedAttr.key, node) }}
            values={values}
            errors={errors}
            cancel={cancel}
            isCancelModelOpen={isCancelModelOpen}
            isFormLocked={isFormLocked}
            isOnEditMode={isEditMode}
          />
        </SecondaryWrapper>
      </PrimaryWrapper>
    </SupplierStepThree>
  );
}

export default SupplierForm;
