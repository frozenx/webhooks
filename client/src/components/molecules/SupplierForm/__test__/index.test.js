import React from "react";
import { shallow } from "enzyme";
import SupplierForm from "../index";

describe("supplier form component test cases", () => {
  let wrapper;
  let mockProps = 
  {
    selectedAttr: [
      {
        name : 'name'
      }
    ],
    partnerName: 'Uni'
  }

  beforeEach(() => {
    wrapper = shallow(<SupplierForm {...mockProps}/>);
  });

  it("render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
