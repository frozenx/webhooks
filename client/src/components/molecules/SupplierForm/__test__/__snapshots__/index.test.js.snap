// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supplier form component test cases render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SupplierForm
    partnerName="Uni"
    selectedAttr={
      Array [
        Object {
          "name": "name",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <BackComponent />,
        <ForwardRef>
          <ForwardRef>
            <h1>
              Uni name
            </h1>
          </ForwardRef>
          <SupplierDetailForm />
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleBackComponent": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <h1>
                Uni name
              </h1>
            </ForwardRef>,
            Array [
              <SupplierDetailForm />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h1>
                Uni name
              </h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Uni name",
              },
              "ref": null,
              "rendered": "Uni name",
              "type": "h1",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": false,
                "rules": Array [
                  [Function],
                  ";",
                  "-webkit-tap-highlight-color:",
                  [Function],
                  ";border:0;box-sizing:border-box;margin:0;outline:",
                  [Function],
                  ";padding:0;position:relative;",
                  ";",
                  "
    margin-top: 34px;
    .input-error {
        color: #cc3333;
    }
",
                ],
              },
              "displayName": "Styled(base-components__RootElement)",
              "foldedComponentIds": Array [
                "base-components__RootElement-sc-150tvch-1",
              ],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "genericFormData": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              [Function],
              ";",
              "-webkit-tap-highlight-color:",
              [Function],
              ";border:0;box-sizing:border-box;margin:0;outline:",
              [Function],
              ";padding:0;position:relative;",
              ";",
              "
    width: 816px;
    margin: 0 auto;
    @media (max-width: 816px) {
        width: 100%;
    }
    form{
        padding-bottom: 24px;
    }
",
            ],
          },
          "displayName": "Styled(base-components__RootElement)",
          "foldedComponentIds": Array [
            "base-components__RootElement-sc-150tvch-1",
          ],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-jTzLTM",
        "isStatic": true,
        "rules": Array [
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-jTzLTM",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <BackComponent />,
          <ForwardRef>
            <ForwardRef>
              <h1>
                Uni name
              </h1>
            </ForwardRef>
            <SupplierDetailForm />
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleBackComponent": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <h1>
                  Uni name
                </h1>
              </ForwardRef>,
              Array [
                <SupplierDetailForm />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <h1>
                  Uni name
                </h1>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Uni name",
                },
                "ref": null,
                "rendered": "Uni name",
                "type": "h1",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "rules": Array [
                    [Function],
                    ";",
                    "-webkit-tap-highlight-color:",
                    [Function],
                    ";border:0;box-sizing:border-box;margin:0;outline:",
                    [Function],
                    ";padding:0;position:relative;",
                    ";",
                    "
    margin-top: 34px;
    .input-error {
        color: #cc3333;
    }
",
                  ],
                },
                "displayName": "Styled(base-components__RootElement)",
                "foldedComponentIds": Array [
                  "base-components__RootElement-sc-150tvch-1",
                ],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "genericFormData": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                [Function],
                ";",
                "-webkit-tap-highlight-color:",
                [Function],
                ";border:0;box-sizing:border-box;margin:0;outline:",
                [Function],
                ";padding:0;position:relative;",
                ";",
                "
    width: 816px;
    margin: 0 auto;
    @media (max-width: 816px) {
        width: 100%;
    }
    form{
        padding-bottom: 24px;
    }
",
              ],
            },
            "displayName": "Styled(base-components__RootElement)",
            "foldedComponentIds": Array [
              "base-components__RootElement-sc-150tvch-1",
            ],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-jTzLTM",
          "isStatic": true,
          "rules": Array [
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-jTzLTM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
